pkg/operation/care/garden_health.go

```diff
@@ -0,0 +1,189 @@
+// Copyright 2023 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package care
+
+import (
+	"context"
+	"time"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/utils/clock"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	v1beta1helper "github.com/gardener/gardener/pkg/apis/core/v1beta1/helper"
+	operatorv1alpha1 "github.com/gardener/gardener/pkg/apis/operator/v1alpha1"
+	resourcesv1alpha1 "github.com/gardener/gardener/pkg/apis/resources/v1alpha1"
+	"github.com/gardener/gardener/pkg/component/etcd"
+	"github.com/gardener/gardener/pkg/component/gardeneraccess"
+	"github.com/gardener/gardener/pkg/component/gardensystem"
+	"github.com/gardener/gardener/pkg/component/hvpa"
+	"github.com/gardener/gardener/pkg/component/istio"
+	"github.com/gardener/gardener/pkg/component/kubecontrollermanager"
+	"github.com/gardener/gardener/pkg/component/kubestatemetrics"
+	"github.com/gardener/gardener/pkg/component/resourcemanager"
+	"github.com/gardener/gardener/pkg/component/vpa"
+	"github.com/gardener/gardener/pkg/features"
+	kubernetesutils "github.com/gardener/gardener/pkg/utils/kubernetes"
+)
+
+var (
+	requiredGardenRuntimeManagedResources = sets.New(
+		etcd.Druid,
+		kubestatemetrics.ManagedResourceName,
+		gardensystem.ManagedResourceName,
+	)
+
+	requiredVirtualGardenManagedResources = sets.New(
+		resourcemanager.ManagedResourceName,
+		gardeneraccess.ManagedResourceName,
+		kubecontrollermanager.ManagedResourceName,
+	)
+)
+
+// GardenHealth contains information needed to execute health checks for garden.
+type GardenHealth struct {
+	garden          *operatorv1alpha1.Garden
+	gardenNamespace string
+	runtimeClient   client.Client
+	clock           clock.Clock
+}
+
+// NewHealthForGarden creates a new Health instance with the given parameters.
+func NewHealthForGarden(garden *operatorv1alpha1.Garden, runtimeClient client.Client, clock clock.Clock, gardenNamespace string) *GardenHealth {
+	return &GardenHealth{
+		runtimeClient:   runtimeClient,
+		garden:          garden,
+		clock:           clock,
+		gardenNamespace: gardenNamespace,
+	}
+}
```

Comment:
Nit: Not sure if we have any consistency with regards to this, but I was shortly confused as the order of the attributes is completely different from the struct definition above.
(The order of the function parameters is yet another order.)